akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -79,11 +79,11 @@ private object PoolConductor {
       }
 
       retryMerge.out ~> slotSelector.in0
-      slotSelector.out ~> doubler ~> route.in
-      retrySplit.out0 ~> flatten ~> slotSelector.in1
-      retrySplit.out1 ~> retryMerge.in1
+      slotSelector.out ~> route.in
+      retrySplit.out(0).filter(!_.isInstanceOf[SlotEvent.RetryRequest]) ~> flatten ~> slotSelector.in1
```

Comment:
Feels clunky, let's eventually add `filterNot`. (`filter(!_.bla)` being a code smell on collections, I react the same when I see it on here)